src/bug-filing/github/github-bug-filing-service.tsx

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import * as _ from 'lodash';
+import { TextField } from 'office-ui-fabric-react/lib/TextField';
+import * as React from 'react';
+
+import { NamedSFC } from '../../common/react/named-sfc';
+import { CreateIssueDetailsTextData } from '../../common/types/create-issue-details-text-data';
+import { BugFilingService } from '../types/bug-filing-service';
+import { SettingsFormProps } from '../types/settings-form-props';
+
+const GitHubBugFilingServiceKey = 'gitHub';
+
+export type GitHubBugFilingSettings = {
+    repository: string;
+};
+
+function buildStoreData(repository: string): GitHubBugFilingSettings {
+    return {
+        repository,
+    };
+}
+
+function isSettingsValid(data: GitHubBugFilingSettings): boolean {
+    return !!data && !_.isEmpty(data.repository.trim());
```

Comment:
- this is going to fail if data is `{}`
- we either need to put a check for `data.repository` or
- we check `isEmpty(data)`

Basically, this is missing some cases. Please add `unit tests` for those cases.

For example: 
- data is `{}`
- data is `null` || `undefined`
- data is not empty but doesn't contain `repository` property